### PR TITLE
Assign

### PR DESCRIPTION
**Pull request by botondhegedus**
app_1           | SELECT cm.id AS cm_id FROM mdl_course_modules cm  LEFT JOIN mdl_course c ON c.id = cm.course  JOIN mdl_assign a ON a.id = cm.instance  JOIN mdl_modules m ON m.id = cm.module  JOIN mdl_user u ON (u.email = 'REDACTED' OR u.username = 'REDACTED')  WHERE m.name = 'assign' AND c.shortname = 'PPD_FirstQuarter2015' AND a.name = 'What is MSF for you?'  

This provides 2 results in my LOCAL NEW DB from 2017-04-01

But by running the script I get this instead:

app_1           |   No course module ID from DB {"id":1976864,"time":1424085174,"userid":185,"ip":"10.111.112.125","course":274,"module":"assign","cmid":24348,"action":"add","url":"view.php?id=24348","info":"3937","u_username":"REDACTED","u_email":"REDACTED","course_shortname":"PPD_FirstQuarter2015","assign_name":"What is MSF for you?"}
app_1           | ERROR for mdl_log.assign.add: insufficient data for query - {"id":1976864,"time":1424085174,"userid":185,"ip":"10.111.112.125","course":274,"module":"assign","cmid":24348,"action":"add","url":"view.php?id=24348","info":"3937","u_username":"REDACTED","u_email":"REDACTED","course_shortname":"PPD_FirstQuarter2015","assign_name":"What is MSF for you?"}

app_1           | completed mdl_log.assign
app_1           | 	mdl_log.assign.add.count: 314
app_1           | 	mdl_log.assign.add.time: 136ms

### CODE REVIEW

File: lib/assign.js [hunk 1]
```diff
@@ -32,16 +32,297 @@ mysql> select action,count(*) from mdl_log where module='assign' group by action
...
+                "WHERE log.module = 'assign'" +
+                "AND log.action = 'add'" +
+                "AND " + restrict_clause,
+
+        sql_old_2pass: (row) => {
+            var m = row.url.match(/id=(\d+)/);
+            if (!m){
+                console.log('No course module ID in URL ' + JSON.stringify(row));
+                return null;
+            }
```
Comment (jlabusch) [inline lib/assign.js]: This looks unfinished... What should it return in the success case?

File: lib/assign.js [hunk 2]
```diff
@@ -32,16 +32,297 @@ mysql> select action,count(*) from mdl_log where module='assign' group by action
...
+                console.log('SELECT cm.id AS cm_id ' +
+                    'FROM mdl_course_modules cm  ' +
+                    'LEFT JOIN mdl_course c ON c.id = cm.course  ' +
+                    'JOIN mdl_assign a ON a.id = cm.instance  ' +
+                    'JOIN mdl_modules m ON m.id = cm.module  ' +
+                    "JOIN mdl_user u ON (BINARY u.email = '" + row["u_email"] + "' OR u.username = '" + row["u_username"] + "')  " +
+                    "WHERE m.name = 'assign' AND c.shortname = '" + row["course_shortname"] + "' AND a.name = '" + row["assign_name"] + "'",
+                    ' \n ',
+                    'No course module ID from DB ' + JSON.stringify(row));
+                return null;
```
Comment (jlabusch) [inline lib/assign.js]: The "return null" statement here is dead code... Did you mean to have console.log()'s return value be the result of sql_match()?

Also, you'll already get the content of row[] printed out by the caller if you return null here, so you might not need console.log() at all.

File: lib/assign.js [hunk 3]
```diff
@@ -32,16 +32,297 @@ mysql> select action,count(*) from mdl_log where module='assign' group by action
...
+        sql_old_2pass: (row) => {
+            var m = row.url.match(/id=(\d+)/);
+            if (!m){
+                console.log('No course module ID in URL ' + JSON.stringify(row));
+                return null;
+            }
+        },
+
+        sql_match: (row) => {
+            return row.cm_id ?
```
Comment (jlabusch) [inline lib/assign.js]: There's no such field in the result of your sql_old query. Do you mean row.cmid?

**Review by jlabusch — CHANGES_REQUESTED** [review]

File: lib/assign.js (hunk 1 above)
Comment (botondhegedus) [reply lib/assign.js]: Fixed

File: lib/assign.js (hunk 2 above)
Comment (botondhegedus) [reply lib/assign.js]: Fixed

File: lib/assign.js (hunk 3 above)
Comment (botondhegedus) [reply lib/assign.js]: Fixed

**Review by jlabusch — APPROVED** [review]

**Pull request merged**